ZeeguuAPI/ZeeguuAPI/ZeeguuAPI.swift

```diff
@@ -0,0 +1,334 @@
+//
+//  ZeeguuAPI.swift
+//  ZeeguuAPI
+//
+//  Created by Jorrit Oosterhof on 28-11-15.
+//  Copyright © 2015 Jorrit Oosterhof. All rights reserved.
+//
+
+import UIKit
+
+/// This class is a gateway to the Zeeguu API. You can use the instance of this class to communicate with the Zeeguu API.
+public class ZeeguuAPI {
+	private static let instance = ZeeguuAPI()
+	
+	var currentSessionID: Int {
+		didSet {
+			let def = NSUserDefaults.standardUserDefaults()
+			def.setObject(currentSessionID, forKey: ZeeguuAPI.sessionIDKey)
+		}
+	}
+	
+	/// Check if a user is logged in.
+	///
+	/// - returns: `true` if a user is logged in, `false` otherwise.
+	public var isLoggedIn: Bool {
+		get {
+			return currentSessionID != 0
+		}
+	}
+	
+	/// Get the `ZeeguuAPI` instance. This method is the only way to get an instance of the ZeeguuAPI class.
+	///
+	/// - returns: The shared `ZeeguuAPI` instance.
+	public static func sharedAPI() -> ZeeguuAPI {
+		return instance;
+	}
+	
+	private init() {
+		self.currentSessionID = 0
+		let def = NSUserDefaults.standardUserDefaults()
+		if (def.objectForKey(ZeeguuAPI.sessionIDKey) != nil) {
+			self.currentSessionID = def.objectForKey(ZeeguuAPI.sessionIDKey)!.integerValue
+		}
+	}
+	
+	/// Registers a user.
+	///
+	/// After the user is registered and logged in, you can use the ZeeguuAPI object to make requests on behalf of the user.
+	///
+	/// - parameter username: The username of the user to register.
+	/// - parameter email: The email address of the user.
+	/// - parameter password: The password of the user.
+	/// - parameter completion: A block that receives a success parameter, which is true if the user was logged in successfully.
+	public func registerUserWithUsername(username: String, email: String, password: String, completion: (success: Bool) -> Void) {
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.AddUser, pathComponents: [email], method: HTTPMethod.POST, parameters: ["username":username, "password":password])
```

Comment:
This is just a very minor point... this line is too long when I'm reading code in GitHub diff... it would be cool if you would format lines for less wide screens. But again, only if you like this.